client/src/modules/Mentor/components/Instructions/Instructions.tsx

```diff
@@ -1,12 +1,42 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Badge, Card, Col, Row, Space, Typography } from 'antd';
 import { INSTRUCTIONS_TEXT, renderDescription, renderSocialLinks } from '.';
+import { DiscordServersApi } from 'api';
+
+interface InstructionsProps {
+  discordServerId: number;
+}
 
 const { Meta, Grid } = Card;
 const { Text } = Typography;
 
-function Instructions() {
-  const { title, description, steps } = INSTRUCTIONS_TEXT;
+const discordServer = new DiscordServersApi();
+
+function Instructions({ discordServerId }: InstructionsProps) {
+  const { title, description } = INSTRUCTIONS_TEXT;
+  const [steps, setSteps] = useState(INSTRUCTIONS_TEXT.steps);
+
+  useEffect(() => {
+    if (!discordServerId) return;
+
+    async function fetchTelegramLink() {
+      const discordServerData = await discordServer.getDiscordServerById(discordServerId);
+
+      const updatedSteps = steps.map(step => {
+        if (!step.links) return step;
+
+        const updatedLinks = step.links.map(link =>
+          link.title === 'telegram' ? { ...link, url: discordServerData.data } : link,
```

Comment:
I don't fully understand logic here: if link is **telegram** link, then we add **discord** data. Why?